SoilMoistureSensor/IoTHubService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Newtonsoft.Json;
+using System.Text;
+
+
+// Service class to communicate with the Azure IoT Hub
+public class IoTHubService
+{
+	private readonly DeviceClient _deviceClient;
+
+	// Creates the service class instance with your IoTHub connection string and your unique device id
+	public IoTHubService(string connectionString)
+	{
+		_deviceClient = DeviceClient.CreateFromConnectionString(connectionString, TransportType.Amqp); // use AMQP
```

Comment:
Optional: could lambda your ctor and probably Resharper format this a bit nice as it's a long line